src/core/request_interceptor.ts

```diff
@@ -0,0 +1,15 @@
+export class RequestInterceptor {
```

Comment:
Could we not set this like we do progressBarDelay? Where it's set on the session, not as a singleton.

Reply:
I think here in Turbo it would be completely possible

Reply:
I was trying to put it in session initially. But we use interceptor in `FetchRequest` class where we don't have an access to the session. Add fetching session from delegates not worked as well since not all delegates have access to the session. That's why I end up using a singleton. Seems like the cleanest solution available.